lib/autoupdate/agent/installer.go

```diff
@@ -175,11 +175,18 @@ func (li *LocalInstaller) Install(ctx context.Context, version, template string,
 	if err != nil {
 		return trace.Errorf("failed to download teleport: %w", err)
 	}
-
 	// Seek to the start of the tgz file after writing
 	if _, err := f.Seek(0, io.SeekStart); err != nil {
 		return trace.Errorf("failed seek to start of download: %w", err)
 	}
+
+	// If interrupted, close the file immediately to stop extracting.
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	go func() {
```

Comment:
nit: `context.AfterFunc(ctx, func() { _ = f.Close() })`

Reply:
Thank you! Totally missed that they added this. So useful!